components/org.wso2.carbon.identity.conditional.auth.functions.user/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/user/GetAuthenticatedAppsFuncImp.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2022, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.conditional.auth.functions.user;
+
+import org.wso2.carbon.identity.application.authentication.framework.config.model.graph.js.JsAuthenticationContext;
+import org.wso2.carbon.identity.application.authentication.framework.context.SessionContext;
+import org.wso2.carbon.identity.application.authentication.framework.dao.UserSessionDAO;
+import org.wso2.carbon.identity.application.authentication.framework.dao.impl.UserSessionDAOImpl;
+import org.wso2.carbon.identity.application.authentication.framework.exception.session.mgt.SessionManagementServerException;
+import org.wso2.carbon.identity.application.authentication.framework.model.Application;
+import org.wso2.carbon.identity.application.authentication.framework.model.UserSession;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Function for retrieving authenticated applications for a given session.
+ */
+public class GetAuthenticatedAppsFuncImp implements GetAuthenticatedApplicationsFunction {
+
+    /**
+     * Retrieve the already authenticated applications for a given session.
+     *
+     * @param context context object.
+     * @return List of already authenticated applications of the given session.
+     */
+    @Override
+    public List<Application> getAuthenticatedApplications(JsAuthenticationContext context) {
+
+        List<Application> applicationList = new ArrayList();
+        String sessionContextKey = context.getWrapped().getSessionIdentifier();
+        try {
+            if (sessionContextKey != null) {
+                UserSessionDAO userSessionDTO = new UserSessionDAOImpl();
```

Comment:
Let's use session service here not DAO and let's get the service from OSGi other than creating a new object here

Reply:
addressed with [https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/103/commits/c4f6cc9dd101db2407d6460189aa11c3ccaf373f](https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/103/commits/c4f6cc9dd101db2407d6460189aa11c3ccaf373f)